Library/Homebrew/missing_formula.rb

```diff
@@ -26,6 +26,10 @@ def blacklisted_reason(name)
           Minimal installation:
             brew cask install basictex
         EOS
+        when "asymptote" then <<~EOS
+          Asymptote is part of MacTeX:
+            brew cask install mactex
```

Comment:
@jmarshall Is this an accurate statement? Thanks!

Reply:
Not especially. s/MacTeX/MacTeX distribution/ or /MacTeX collection/ would help. But mostly it's incomplete as to the casks available (as previously noted in https://github.com/Homebrew/brew/pull/5816#discussion_r261182159) and to choices made during pkg installation as to how much of the collection actually gets installed.

Reply:
@jmarshall As I think I've explained above: if a new entry is added for Asymptote I want the user to be provided with a `brew cask` invocation that will install it. If `brew cask install mactex` will not do so: what will?

Reply:
As I've explained above, the existing _missing_formula.rb_ entry for `tex`/`latex`/etc already contains an accurate list of the TeX-related casks available. The single `cask install` you've added is incomplete information; something like what I've pushed would be more accurate (you might prefer `brew search`).

As a Brew user looking for Asymptote, what I actually needed was (1) a pointer to the MacTeX that I already have installed, (2) a hint that the command is called `asy`, and (3, if I were new to MacTeX and/or had installed a cut-down MacTeX subset) a pointer to TeX Live Utility. What I didn't need was being reminded how to type `brew [cask] install mactex`.

Hence the PR as at c508bda668655153691393279bde511e00c03140. However (2) and (3) might be considered out of Brew's scope, so if something like the current bfee3be772fa1bd224fea2da35e67d81a3b82bb8 makes the Homebrew maintainers happy that is fine.

Reply:
I'll try to be explicit about what is needed:
- I need this PR to point to a single `brew cask install` which will install Asymptote for it to be merged.
- I need confirmation on whether `brew cask install mactex` will install Asymptote? 

If you're unwilling or unable to do either of these: please close this PR, thanks.

Reply:
Could you please explain what is wrong with the PR displaying a single `brew install mactex` command that gives the user the information they need to choose between the available casks?

Reply:
It doesn't make clear which of the casks includes or doesn't include Asymptote so it's an unhelpful message.

Reply:
It would have been helpful to be explicit that that was what your concern was about.

Asymptote is installed by the `mactex` and `mactex-no-gui` casks, but not by `basictex`. (It is also of course installed by `basictex`+`tlmgr install asymptote` or by installing the upstream MacTeX package, which are options that had been covered by previous versions of this PR.)

So if the Homebrew maintainers want to adjust this to some variant of `brew cask install mactex{,-no-gui}` on merging, please do so.

Reply:
> If you're unwilling or unable to do either of these: please close this PR, thanks.

Reply:
Closing this PR as a result. This has been a frustrating experience; please address maintainers feedback without arguing in hopeful future PRs.

Reply:
> If you're unwilling or unable to do either of these: please close this PR, thanks.

I spent several hours doing those things yesterday, pal.

* You completely rewrote the text of this PR once, so I anticipated you would want to do so again with your preferred `cask` text.

* I spent several hours breaking my TeX installation to confirm which casks contained Asymptote.

Thank you for your time. If I should happen to raise future PRs, I hope they will be less frustrating experiences for all concerned.

Reply:
> I spent several hours doing those things yesterday, pal.
> I need this PR to point to a single brew cask install which will install Asymptote for it to be merged.

You did not do that.

> * You completely rewrote the text of this PR once, so I anticipated you would want to do so again with your preferred `cask` text.

I've restored it to that text as it is accurate (based on what you said above) and meets the requirements (a single `brew cask install` invocation).

> Thank you for your time.

And for yours.

> If I should happen to raise future PRs, I hope they will be less frustrating experiences for all concerned.

The documentation helps cover some of this: 
> Do not argue with Homebrew maintainers. You may disagree but unless they change their mind, please implement what they request. Ultimately they control what is included in Homebrew, as they have to support any changes that are made.
https://github.com/Homebrew/brew/blob/master/docs/How-To-Open-a-Homebrew-Pull-Request.md